### PR TITLE
Replace iD global with iD.Context() module

### LINKED ISSUE #3179: ES6 Modules: phase 2
Phase 1 of iD's modularization is captured in https://github.com/openstreetmap/iD/issues/3118 and completed by our lovely community! Phase 2 is one task that @kepta is leading out and @bhousel is helping land:

So, phase 1 involved a lot of work and was split by the natural split points in iD: each module, like iD.ui and iD.svg. We have some degree of 'using modules for requiring their dependencies' but not 100% - the modules in `modules/*` still rely on the iD global all over the place. And we're creating a bundle per part of iD.

Phase 2 means

* Remove the iD global for the eslint configuration of `modules/*`, so that relying on global iD is forbidden
* Using `import` instead to require dependencies
* Instead of making a bunch of bundles, create one entry file that imports _all_ of iD, and goes through one rollup run to create the iD bundle.

---

I'll open a ticket for a moment for phase 3; the two tasks are/should be separable so that we can keep shipping little bits of work at a time.

### PR DESCRIPTION
**Pull request by bhousel**
This PR replaces the iD global window object with a Context module.

@kepta, @tmcw Does this seem right?  This might close #3179 

(though we still have to do something with `iD.data` / `data_dev.js`)

### CODE REVIEW

File: modules/core/context.js [hunk 1]
```diff
@@ -328,15 +340,16 @@ window.iD = function () {
 
 
     /* Init */
+    context.version = '2.0.0-alpha.1';
```
Comment (kepta) [inline modules/core/context.js]: excited

**Comment by kepta** [comment]
@bhousel looks good to me.
Lets merge it already !!

**Comment by tmcw** [comment]
👍 Tested, LGTM. Doesn't close #3179 yet - we'll need to remove iD from the eslint globals in order to lock that down.

**Comment by bhousel** [comment]
> we'll need to remove iD from the eslint globals in order to lock that down.

ok, I'll take care of `.eslintrc`

Also what should we do about `iD.data`?
I know we import/export the .json files too, so my plan was to put an `index.js` in the data folder that does that.

**Pull request merged**